src/components/main/main.jsx

```diff
@@ -149,12 +172,24 @@ function Main1() {
       })
     );
   }
+}
 
   function joinGameHandler(gameKey, playerName) {
     db.collection("games")
       .where("gameKey", "==", gameKey)
       .get()
-      .then((querySnapShot) => {
+      .then((querySnapShot) => {  
+        if(playerName==""){
+          setplayerNameMessage("enter valid player name");
+        }
+        else{
+        setplayerNameMessage("");
+        }
+        if(querySnapShot.docs.length<=0 || gameKey==""){
```

Comment:
This gameKey=="" condition should be moved to the top, so that we don't even query database if the user doesn't enter  anything. On top of the function , condition should be like this:
if(gameKey=="")
return;